docs/modules/user-guide/partials/proc_defining-endpoints.adoc

```diff
@@ -29,113 +29,73 @@ components:
       mountSources: true
       command: ['sleep', 'infinity']
       env:
-        - name: GOPATH
+        - name: gopath
           value: $(PROJECTS_ROOT)/go
-        - name: GOCACHE
+        - name: gocache
           value: /tmp/go-cache
       endpoints:
       - name: web
         targetPort: 8080
-        configuration:
-          discoverable: false
-          public: true
-          protocol: tcp
-          scheme: http
+        exposure: public
   - name: postgres
     container:
       image: postgres
       memoryLimit: 512Mi
       env:
-        - name: POSTGRES_USER
+        - name: postgres_user
           value: user
-        - name: POSTGRES_PASSWORD
+        - name: postgres_password
           value: password
-        - name: POSTGRES_DB
+        - name: postgres_db
           value: database
       endpoints:
         - name: postgres
           targetPort: 5432
-          configuration:
-            discoverable: true
-            public: false
+          exposure: none
 ----
 ====
 
-Here, there are two container, each defining a single endpoint. Endpoint is an accessible port that can be made accessible inside the workspace or publicly (example, from the UI). Each endpoint has a name and port, which is the port on which certain server running inside the container is listening. The following are a few attributes that you can set on the endpoint:
+Here, there are two container, each defining a single endpoint. Endpoint is an accessible port that can be made accessible inside the workspace or publicly, for example, from the UI. Each endpoint has a name and port, which is the port on which certain server running inside the container is listening. The following are a few attributes that you can set on the endpoint:
```

Comment:
We might want to place a TODO comment here to come back and take another look at the language use and style issues in this section. There are a lot of issues here. Just something to note. :)

Reply:
Will address this concern in  a PR fo the newly filed issue, #393: https://github.com/devfile/api/issues/393